pkg/kubectl/cmd/exec.go

```diff
@@ -159,6 +167,8 @@ func (p *ExecOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn []s
 		}
 	}
 
+	p.StreamTimeout = cmdutil.GetFlagDuration(cmd, "stream-timeout")
```

Comment:
@smarterclayton or @fabianofranz would it make more sense from a user's perspective to just re-use the value from the global `--request-timeout` flag, or is it worth adding a third timeout option?